core/sys/posix/dirent.odin

```diff
@@ -193,11 +193,41 @@ when ODIN_OS == .Darwin {
 } else when ODIN_OS == .NetBSD {
 
 	dirent :: struct {
-		d_ino:     ino_t,                    /* [PSX] file number of entry */
-		d_reclen:  c.uint16_t,               /* length of this record */
-		d_namelen: c.uint16_t,               /* length of string in d_name */
-		d_type:    D_Type,                   /* file type  */
-		d_name:    [512]c.char `fmt:"s,0"`,  /* [PSX] entry name */
+		d_ino:     ino_t,                   /* [PSX] file number of entry */
+		d_reclen:  c.uint16_t,              /* length of this record */
+		d_namelen: c.uint16_t,              /* length of string in d_name */
+		d_type:    D_Type,                  /* file type  */
+		d_name:    [512]c.char `fmt:"s,0"`, /* [PSX] entry name */
+	}
+
+} else when ODIN_OS == .Linux {
+
+	when ODIN_ARCH == .i386 || ODIN_ARCH == .wasm32 || ODIN_ARCH == .arm32 {
+
+		dirent :: struct {
+			d_ino: ino_t32,                  /* [PSX] file number of entry */
+			d_off: off_t32,                  /* directory offset of the next entry */
+			d_reclen: c.uint16_t,            /* length of this record */
+			d_type: D_Type,                  /* file type  */
+			d_name: [256]c.char `fmt:"s,0"`, /* [PSX] entry name */
+		}
+	} else when ODIN_ARCH == .amd64 || ODIN_ARCH == .wasm64p32 || ODIN_ARCH == .arm64 {
+
+		dirent :: struct {
+			d_ino: ino_t,                    /* [PSX] file number of entry */
+			d_off: off_t,                    /* directory offset of the next entry */
+			d_reclen: c.uint16_t,            /* length of this record */
+			d_type: D_Type,                  /* file type  */
+			d_name: [256]c.char `fmt:"s,0"`, /* [PSX] entry name */
+		}
+	}
+
+	dirent64 :: struct {
```

Comment:
I was following the `dirent.h` definitions on Linux, on my system this options exists (see below). But if it's still not needed because it's actually not POSIX-specified, I'll remove it.

![image](https://github.com/user-attachments/assets/2f2b401d-2dd2-4e96-ab9c-b145509561ee)

Reply:
Yeah the `*64` types are meant for 32-bit platforms that want to use 64-bit functions and types, if it's not posix specified, Odin doesn't really need that because all Odin platforms support 64-bit types and we don't need to provide it as an option, we can just force 64-bit for all of these structs.

Reply:
Right, we don't do anything in this package that isn't posix specified.